### ISSUE TITLE
Add GH action to find broken links in Documentation (and elsewhere?) 

### ISSUE THREAD

**Issue by mccabete** [PecanProject/pecan#3342]
To put in a test that finds broken links, we could add a GH Action like https://github.com/marketplace/actions/lychee-broken-link-checker to check for broken links on build; it isn't clear to me where to put this in the build process.

---

Original bug report: 

I've noticed that most/all of the links in the documentation that are shortcuts to other parts of the documentation tend to throw a 404 error.  This doesn't seem to be true of links that are to external places. 

## Example

[Documentation](https://pecanproject.github.io/pecan-documentation/master/logging.html#outputs): "Adding-an-Input-Converter" redirects to [this](https://pecanproject.github.io/pecan-documentation/master/Adding-an-Input-Converter.md). 

## Next Steps
1. figure out what makes these links out of date
2. put in a test that catches expired links

**Comment by mccabete** [comment]
@tonygardella, not sure if I should move this to pecan

**Comment by mdietze** [comment]
Yes, please move to the general pecan repo as we no longer keep the documentation in a separate repo

**Comment by mdietze** [comment]
@tonygardella these appear to be things that really need to be fixed in this release

**Comment by dlebauer** [comment]
The links in the original bug report post are broken, but it seems like a lot (prob. not all!) of this has been fixed.

To put in a test that finds broken links, we could add a GH Action like https://github.com/marketplace/actions/lychee-broken-link-checker to check for broken links on build; it isn't clear to me where to put this in the build process.

**Comment by infotroph** [comment]
I support automated link checking, with the caveat that any "hit a bunch of external links" task is going to be slower than we think and will see lots of transient failures even from basically-working sites. 

I vote we put this outside the build process, running on a set schedule (weekly?) rather than every PR and opening issues for (repeated?) failures rather than breaking the build. 
Operationally, this means it should be implemented in a new file named something like `.github/workflows/check-links.yml` and its first 10 lines should be similar to `.github/workflows/ci-weekly.yml`.

**Comment by Sweetdevil144** [comment]
Would this help : #3269 ?
